Rewrite the 'Get started' instructions (#259)
signalfx/signalfx-dotnet-tracing: docs/README.md

## Get started

> Make sure  you set up the Splunk OpenTelemtry Collector

Reviewer: Consider link to instruction how to do it.

> Make sure you set up the [Splunk OpenTelemtry Collector](https://github.com/signalfx/splunk-otel-collector)

Reviewer 2: ```suggestion
Make sure you set up the [Splunk OpenTelemtry Collector](https://quickdraw.splunk.com/redirect/?product=Observability&location=windows.collector.setup&version=current)
```

Better to link to the official docs. We use quickdraw links to better manage redirections.

Author: Any why not https://docs.splunk.com/Observability/gdi/opentelemetry/install-windows.html#nav-Install-on-Windows ?
I will create a Jira story to update some hyperlink so that they link to the official docs. We can also have a split between "our" `/docs/README.md` and "user-facing" `README.md`

Reviewer 2: The problem with direct links like that is that they're hard to update and keep track of. That's why we use redirects. But yeah, better to use direct links than links to GitHub READMEs.

Author: @theletterf Take notice that the the hyperlink I put directs to something else 

- "Docs » Collect infrastructure metrics and logs » Collect Windows data"

vs

- "Docs » Install and configure Splunk Distribution of OpenTelemetry Collector » Install the Splunk OpenTelemetry Collector » Install on Windows"

Reviewer 2: Let's use direct links for now then!

Author: We will try to address it later in a more global way.

to receive telemetry data.

### Installation

You can find the latest installation packages on the

> [Releases](https://github.com/signalfx/signalfx-dotnet-tracing/releases/lastest)

Reviewer: ` ` ` suggestion
[Releases](https://github.com/signalfx/signalfx-dotnet-tracing/releases/latest)
` ` `

page.

> | Sufix         | Operating System    | Architecture | Installing via Command Line | Notes |

Reviewer 2: ```suggestion
| File         | Operating system    | Architecture | Install command | Notes |
```

| ---           | ---                 | ---          | ---          | ---   |
| `x86_64.rpm`  | Red Hat-based Linux distributions | x64 | `rpm -ivh signalfx-dotnet-tracing.rpm` | RPM package |
| `x64.msi`     | Windows 64-bit | x64 |  `msiexec /i signalfx-dotnet-tracing-x64.msi /quiet` | |
| `x86.msi`     | Windows 32-bit | x86 | `msiexec /i signalfx-dotnet-tracing-x86.msi /quiet` | |

> | `musl.tar.gz` | x64 Linux distributions using [qlibc](https://wiki.musl-libc.org/projects-using-musl.html) | x86 | `tar -xf signalfx-dotnet-tracing.tar.gz -C /` | Currently, all [officially supported Linux distribtions](https://docs.microsoft.com/dotnet/core/install/linux) but Alpine uses glibc |

Reviewer: _ut Alpine uses glibc_  It is not clear for me what that's mean.

Author: I will change to `except` 👍

Author: Check cd74343d

> | `musl.tar.gz` | x64 Linux distributions using [qlibc](https://wiki.musl-libc.org/projects-using-musl.html) | x86 | `tar -xf signalfx-dotnet-tracing.tar.gz -C /` | Currently, all [officially supported Linux distribtions](https://docs.microsoft.com/dotnet/core/install/linux) but Alpine uses glibc |

Reviewer: x64 Linux distributions using or x86? next column contains x86

Author: x64  - nice catch

Author: also I removed the `musl` from the first column

| `amd64.deb`   | Debian-based Linux distributions | x64 | `dpkg -i signalfx-dotnet-tracing.debm` | DEB package |
<!-- TODO: | `musl.tar.gz` | x64 Linux distributions using [musl](https://wiki.musl-libc.org/projects-using-musl.html) | x64 | `tar -xf signalfx-dotnet-tracing-musl.tar.gz -C /` | Alpine Linux uses musl | -->

> :warning: On Linux, after the installation, you need to additionally run:

Reviewer 2: All distros? Wasn't this step optional?

Author: It was but as far as I see it should not be optional.

Reviewer 2: If it's a required step, let's remote `:warning:` and leave it as a regular step.

Author: Addressed https://github.com/signalfx/signalfx-dotnet-tracing/pull/259/commits/8bf18465707d07a23fa190a11d4eb75fa6c470f9

```bash
/opt/signalfx/createLogPath.sh
```

> ### Instrument an application on Windows

Reviewer 2: ```suggestion
### Instrument a .NET application on Windows

To instrument a .NET application on Windows, run the following commands:
```

Author: Just double-checking - are you sure? Everything here is about .NET applications. Should we e.g. then change the other tit;es to  `### Instrument a .NET web application deployed on ISS` and `### Instrument Windows Service hosting a .NET application` ?

Reviewer 2: Doesn't hurt to be specific. I like your heading proposals!

Author: Addressed https://github.com/signalfx/signalfx-dotnet-tracing/pull/259/commits/6fdf5410aa7e91372ca99a2e4dac93ae26f2a98f

```powershell
$Env:COR_PROFILER = "{B4C89B0F-9908-4F73-9F59-0D77C5A06874}"      # Select the .NET Framework Profiler
$Env:COR_ENABLE_PROFILING = "1"                                   # Enable the .NET Framework Profiler
$Env:CORECLR_PROFILER = "{B4C89B0F-9908-4F73-9F59-0D77C5A06874}"  # Select the .NET (Core) Profiler

> $Env:CORECLR_ENABLE_PROFILING = "1"                               # Enable the .NET (Core) Profiler

Author: as an improvement, we could create a `.ps1` script out of it (equivalent for `defaults.env`) and then the installer could put it here: `%ProgramData%\SignalFx .NET Tracing\defaults.ps1`

Reviewer: Phase 3 jira item.

Author: Want to know what are the preferences. Maybe it is "Won't do". @theletterf would it be any better if we add such script?

Author: Done

> $Env:CORECLR_ENABLE_PROFILING = "1"                               # Enable the .NET (Core) Profiler

Reviewer 2: Does it have to be `Env`? Can it be just `env`?

Author: `PascalCase` is the convention that is used in PowerShell 

Also this is how the official docs are recommending: https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_environment_variables?view=powershell-7.2

Reviewer 2: True! It's just that for most of our GDI UI snippets we went lowercase. But I know PS is rather tolerant of casing inconsistencies.

Author: It is. But it would decrease my PowerShell karma which I am afraid to lose 😋

> # now the auto-instrumentation is configured in this shell session
> # you can set additional settings and run your application e.g.

Reviewer 2: ```suggestion
# Now the autoinstrumentation is configured in this shell session.
# You can set additional settings and run your application, for example:
```

$Env:SIGNALFX_SERVICE_NAME = "my-service-name"                    # Set the service name
dotnet run                                                        # Run your application                                                     
```

> ### Instrument an application on Linux

Reviewer 2: ```suggestion
### Instrument a .NET application on Linux

To instrument a .NET application on Linux, run the following commands:
```

Author: Addressed https://github.com/signalfx/signalfx-dotnet-tracing/pull/259/commits/6fdf5410aa7e91372ca99a2e4dac93ae26f2a98f

```bash
source /opt/signalfx/defaults.env               # Enable the .NET (Core) Profiler

> # now the auto-instrumentation is configured in this shell session
> # you can set additional settings and run your application e.g.

Reviewer 2: ```suggestion
# Now the autoinstrumentation is configured in this shell session.
# You can set additional settings and run your application, for example:
```

export SIGNALFX_SERVICE_NAME="my-service-name"  # Set the service name
dotnet run                                      # Run your application 
```

> ### Instrument a Windows Service

Reviewer 2: ```suggestion
### Instrument a Windows Service

To instrument a Windows service, run the following commands:
```

Author: We have a story to update the whole section

<!-- TODO:

Update this section to use a PowerShell script that sets all of the following env vars:
COR_PROFILER={B4C89B0F-9908-4F73-9F59-0D77C5A06874}
COR_ENABLE_PROFILING=1
CORECLR_PROFILER={B4C89B0F-9908-4F73-9F59-0D77C5A06874}
CORECLR_ENABLE_PROFILING=1
-->

> Enable instrumentation for a specific Windows service:

Reviewer 2: ```suggestion
Then, enable the instrumentation for the service by running the following command:
```

Author: We have a Jira story to update the whole section

   reg add HKLM\SYSTEM\CurrentControlSet\Services\<ServiceName>\Environment /v COR_ENABLE_PROFILING /d 1
   ```

> - For .NET or .NET Core applications:

Reviewer 2: This step is for Windows services, for applications, or for both?

Author: Windows Services that are hosting .NET applications 😉
